src/Protocol/ATProtocol.php

```diff
@@ -357,7 +357,7 @@ public function getDid(string $handle): string
 	 */
 	public function getDidByProfile(string $url): string
 	{
-		if (preg_match('#^' . self::WEB . '/profile/(.+)#', $url, $matches)) {
+		if (preg_match('#^' . self::WEB . '/profile/(.+)#', $url, $matches) && is_array($matches)) {
```

Comment:
What's the case where `preg_match` returns a truthy value but `$matches` isn't filled with matches? Even if there aren't any capturing group in the pattern, `$matches[0]` contains the input string in case of a match.

Reply:
I'm not sure what is causing this, but I had the mentioned warning on three machines. Yes, this shouldn't happen, but it does.

Reply:
If preg_match has a problem (for instance with the regex) it returns `false` and `$matches` will not be set, ergo not an array.

```php
preg_match('/', '', $matches);
//$matches will be unset
```

Reply:
Sure, but we didn't interact with `$matches` if `preg_match()` return value was falsy.

@annando What was the exact line of the warning?

Reply:
See the quoted text in the pull request description.

Reply:
Warning text says its not in this file but in src/Model/contact (3365)

Reply:
Oh yeah, I completely copied the wrong log line ...

Reply:
So what is the correct log line of the associated error message?

Reply:
It's the line that I quoted in the pull request.

Reply:
Yes, and it's incomplete, missing the file name and line number, which I'm asking about.

Reply:
Well, it's that line. I don't have the original entry anymore, since I delete all old logs after two days.

Reply:
If it’s the condition line that triggered the error, then I believe we should simply initialize the `$matches` variable.
```suggestion
		$matches = [];
		if (preg_match('#^' . self::WEB . '/profile/(.+)#', $url, $matches)) {
```

Reply:
That would create an error as well, since we address an element in that array and that element doesn't exist.

Reply:
If the error is on a line that contains `$matches[1]` then it is indeed not the correct solution, but if the error is on the condition line itself where we don't access an array offset directly, then I believe the initialization would be enough. I still feel like your solution, while working, is redundant and misleading, and I'd be really curious what values are triggering the original warning so that we could reliably reproduce it.

Reply:
I've been throwing a bunch of stuff on the 3v4l wall, see what sticks, but so far in similar code conditions I've not been able to get the same Notice message: https://3v4l.org/6nj9n#v8.2.27

This means I've not needed the same fix you submitted. I'm still really puzzled as to why we need it.